lib/storage/src/issues_subscribers.rs

```diff
@@ -2,40 +2,42 @@ use std::sync::Arc;
 
 use collection::events::{CollectionDeletedEvent, IndexCreatedEvent, SlowQueryEvent};
 use issues::broker::Subscriber;
+use issues::Code;
+use segment::problems::UnindexedField;
 
-use crate::content_manager::toc::TableOfContent;
-use crate::dispatcher::Dispatcher;
-use crate::rbac::Access;
-
-#[allow(dead_code)] // TODO: remove
-#[derive(Clone)]
-pub struct UnindexedFieldSubscriber {
-    toc: Arc<TableOfContent>,
-    access: Access,
-}
-
-impl UnindexedFieldSubscriber {
-    pub fn new(dispatcher: Arc<Dispatcher>) -> Self {
-        let access = Access::full("For Issues API");
-        let toc = dispatcher.toc(&access).clone();
-        Self { toc, access }
-    }
-}
+#[derive(Clone, Copy)]
+pub struct UnindexedFieldSubscriber;
 
 impl Subscriber<SlowQueryEvent> for UnindexedFieldSubscriber {
-    fn notify(&self, _event: Arc<SlowQueryEvent>) {
-        // TODO: process filters and compare against the schema taken from collection info
+    fn notify(&self, event: Arc<SlowQueryEvent>) {
+        if event.filters.is_empty() {
+            return;
+        }
+
+        tokio::spawn(async move {
+            for filter in &event.filters {
+                segment::problems::UnindexedField::submit_possible_suspects(
+                    filter,
+                    &event.schema,
+                    event.collection_id.clone(),
+                )
+            }
+        });
```

Comment:
The inner function is synchronous and doesn't have any `.await`s. Does it make sense to put it into `tokio::spawn`?

Reply:
You're right, it previously did some async work but not anymore, and the processing it does now is not so expensive. We can remove it